### ISSUE TITLE
Common Error base?

### ISSUE THREAD

**Issue by ajford** [PyTables/PyTables#549]
How feasible would it be to create a common Tables.TablesError base class?

I've been writing a few quick test scripts and tools using PyTables, and I'd like to do a broad catch on any PyTables exceptions, just in case I missed a corner case or something. But I don't want to just catch all, as I might hide some other problem. 

In other packages I've used, this is easy to do as all errors are subclasses of a base package exception class, with more specific mix-ins if desired (i.e. ValueError or RuntimeError). However, nothing of this sort is done with PyTables.

It's a simple enough fix, so I'd be willing to do a PR for this if there's interest.

**Comment by FrancescAlted** [comment]
Yes, I think that would be a valuable contribution.  Thanks for willing to contribute.
